src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -97,7 +97,7 @@ protected function doSave(array $values, int $lifetime, array $addTagData = [],
         if ('noeviction' !== $eviction && 0 !== strpos($eviction, 'volatile-')) {
             CacheItem::log($this->logger, sprintf('Redis maxmemory-policy setting "%s" is *not* supported by RedisTagAwareAdapter, use "noeviction" or  "volatile-*" eviction policies', $eviction));
 
-            return false;
+            return [];
```

Comment:
`return array_keys($values);`

Reply:
Sorry, not sure about this. I guess it still should return an empty array because of https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/AbstractAdapter.php#L160.

When somebody switches redis to non-supported eviction policy, https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php#L98 error will be logged. But no data will be stored in the redis. And it is expected behavior. Please correct me if I`m wrong.

Reply:
200% sure this must return what I suggested. The contract is to return the list of failed ids. Returning the empty arrays means "save went fine", which is not the case.

Reply:
Maybe a better option would be to throw actually, instead of the log line.
AbstractAdapter will take care of turning the exception into a log message;

Reply:
Ok, made https://github.com/symfony/symfony/pull/36606/commits/af59e62cdcc8b1a5dbc27e60c10df06a78652dcf

Reply:
Throwing an exception will have the same result as returning an empty array, isn't it?

Reply:
I`m just checking https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/AbstractAdapter.php#L155-L176

Reply:
Ah ... throwing an exception will lead to retry to store the values?

Reply:
> Throwing an exception will have the same result as returning an empty array, isn't it?

absolutely not

>  Ah ... throwing an exception will lead to retry to store the values?

yes, same as returning the list of ids.
retries will fail again, and will generate a lig line with the details, that's what we want.

Reply:
Sorry, just to make sure. In the test scenario described in this PR:
1. If empty array will be returned:
- error about the unsupported redis eviction policy will be logged
- cache times will be ignored without any further logs/details
2. if an exception is thrown:
- error about the unsupported redis eviction policy will be logged
- items will be retried to store (not sure if there is a reason to do it)
- details about each item will be logged

And the second option is the correct? What exception should be thrown in this case? Is `Symfony\Component\Cache\Exception\CacheException` fine?

Reply:
Also:
1. the save() method will return `true` (this would be wrong)
2. the save() method will return `false` (this is correct)

A `LogicException` from the component would fit best I think.

Reply:
Done in https://github.com/symfony/symfony/pull/36606/commits/3d6e942da557a44bf9bbc17eef8efc31d1e2141b.
There were some failing checks before, but seems like they are not related to this PR.